packages/hooks/use-theme/src/index.ts

```diff
@@ -0,0 +1,81 @@
+import {useCallback, useEffect, useState} from "react";
+
+// constant properties for Theme
+export const ThemeProps = {
+  // localStorage key for storing the current theme
+  KEY: "nextui-theme",
+  // light theme
+  LIGHT: "light",
+  // dark theme
+  DARK: "dark",
+  // system theme
+  SYSTEM: "system",
+} as const;
+
+// type definition for Theme using system theme names or custom theme names
+export type customTheme = string;
+export type Theme =
+  | typeof ThemeProps.LIGHT
+  | typeof ThemeProps.DARK
+  | typeof ThemeProps.SYSTEM
+  | customTheme;
+
+/**
+ * React hook to switch between themes
+ *
+ * @param defaultTheme the default theme name (e.g. light, dark, purple-dark and etc)
+ * @returns An object containing the current theme and theme manipulation functions
+ */
+export function useTheme(defaultTheme?: Theme) {
+  const [theme, setThemeState] = useState<Theme>(() => {
+    let storedTheme: Theme | undefined;
+
+    try {
+      storedTheme = localStorage.getItem(ThemeProps.KEY) || undefined;
+    } catch {}
+
+    // return stored theme if it is selected previously
+    if (storedTheme) return storedTheme;
+
+    // if it is using system theme, check `prefers-color-scheme` value
+    // return light theme if not specified
+    if (defaultTheme === ThemeProps.SYSTEM) {
+      return window.matchMedia && window.matchMedia("(prefers-color-scheme: dark)").matches
+        ? ThemeProps.DARK
+        : ThemeProps.LIGHT;
+    }
+
+    // return default theme with light theme as default one
+    return defaultTheme ?? ThemeProps.LIGHT;
```

Comment:
It would be more convenient for users if the system theme is applied when the theme is not found in localStorage or no defaultTheme is provided. Next.js docs follows a similar approach.
https://nextjs.org/docs

Reply:
If modified as you said, when `defaultTheme` is not passed, `defaultTheme` will default to `system`. At this point, the result is returned in line 43, and this line only needs to `return defaultTheme`, no additional `??` needed.